ui/src/views/Activate/Syncs/SyncRecords/SyncRecords.tsx

```diff
@@ -0,0 +1,196 @@
+
+const SyncRecords = (): JSX.Element => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { syncId, syncRunId } = useParams();
+  const toast = useCustomToast();
+
+  const pageId = searchParams.get('page');
+  const [currentPage, setCurrentPage] = useState(Number(pageId) || 1);
+
+  useEffect(() => {
+    setSearchParams({ page: currentPage.toString() });
+  }, [currentPage, setSearchParams]);
+
+  const [currentFilter, setCurrentFilter] = useState<'success' | 'failed'>('success');
+
+  const {
+    data: syncRunRecords,
+    isLoading: isSyncRecordsLoading,
+    isError: isSyncRecordsError,
+  } = useQuery({
+    queryKey: ['activate', 'sync-records', syncRunId, currentPage],
+    queryFn: () => getSyncRecords(syncId as string, syncRunId as string, currentPage.toString()),
+    refetchOnMount: true,
+    refetchOnWindowFocus: false,
+  });
+
+  const filteredRecords = useMemo(
+    () => syncRunRecords?.data.filter((record) => record.attributes.status === currentFilter),
+    [syncRunRecords, currentFilter],
+  );
+
+  const syncRunRecordColumns = useMemo(
+    () =>
+      filteredRecords && filteredRecords.length > 0
+        ? Object.keys(filteredRecords[0].attributes.record)
+        : [],
+    [filteredRecords],
+  );
+
+  const SYNC_RUNS_RECORDS_COLUMNS = useMemo(
+    () => [
+      {
+        key: 'status',
+        name: 'Status',
+      },
+      ...syncRunRecordColumns.map((column) => ({
+        key: column,
+        name: column,
+      })),
+    ],
+    [syncRunRecordColumns],
+  );
+
+  const tableData = useMemo(() => {
+    const rows = filteredRecords?.map((data) => {
+      return SYNC_RUNS_RECORDS_COLUMNS.reduce(
+        (acc, { key }) => ({
+          ...acc,
+          [key]: <TableItem field={key} data={data} />,
+          id: data.id,
+        }),
+        {},
+      );
+    });
+
+    return {
+      columns: SYNC_RUNS_RECORDS_COLUMNS,
+      data: rows || [],
+      error: '',
+    };
+  }, [filteredRecords, SYNC_RUNS_RECORDS_COLUMNS]);
+
+  const handleNextPage = () => {
+    if (syncRunRecords?.links?.next) {
+      setCurrentPage((prevPage) => prevPage + 1);
+    }
+  };
+
+  const handlePrevPage = () => {
+    if (syncRunRecords?.links?.prev) {
+      setCurrentPage((prevPage) => Math.max(prevPage - 1, 1));
+    }
+  };
+
+  const FilterTabs = () => {
+    return (
+      <TabList gap='8px'>
+        <TabItem
+          text='Successful'
+          action={() => setCurrentFilter('success')}
+          isBadgeVisible
+          badgeText={syncRunRecords?.data
+            .filter((record) => record.attributes.status === 'success')
+            .length.toString()}
+        />
+        <TabItem
+          text='Failed'
+          action={() => setCurrentFilter('failed')}
+          isBadgeVisible
+          badgeText={syncRunRecords?.data
+            .filter((record) => record.attributes.status === 'failed')
+            .length.toString()}
+        />
+      </TabList>
+    );
+  };
```

Comment:
can we move this outside of this component?
because for every re-render, this component will be created again and again which I don't think so is needed.